src/Silverback.Integration/Messaging/IEndpoint.cs

```diff
@@ -49,6 +50,12 @@ public interface IEndpoint
         /// </summary>
         string? FriendlyName { get; }
 
+        /// <summary>
+        ///     Gets or sets the validation mode. This option can be used to determines if validation has to be performed and if
+        ///     should throw an exception.
+        /// </summary>
+        ValidationMode ValidationMode { get; }
```

Comment:
`MessageValidationMode` is maybe a better name, more precise. What do you think?

Reply:
The default is `ValidationMode.None`, making this an `opt-in` feature.

I wonder if we shouldn't set it to `WithWarning` per default. Like this we are technically backward compatible but people will start seeing the warnings.
I'm unsure because of the performance hit. By the way...how big is that? Could we measure it via a benchmark to have an idea? (see benchmarks project)